WebKit LLVM branch cannot build WebKit for iOS simulator

rniwa opened the pull request:
#### d4b79a9469fb94dae741d6c0d9325d3955035773
<pre>
WebKit LLVM branch cannot build WebKit for iOS simulator
<a href="https://bugs.webkit.org/show_bug.cgi?id=300293">https://bugs.webkit.org/show_bug.cgi?id=300293</a>

Reviewed by David Kilzer and Mike Wyrzykowski.

Make WebKit compatible with webkit branch of LLVM.

* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::maxIndexValue): Ignore &quot;-Wpsabi&quot; warning.
* Source/WebKit/Configurations/BaseExtension.xcconfig:
Set CLANG_MODULE_CACHE_PATH to an empty string to avoid hitting
an error in pre-compiling WebKit2Prefix.h.

Canonical link: <a href="https://commits.webkit.org/301156@main">https://commits.webkit.org/301156@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b70a17c1ec977f6018ea59c43ac508dec9e79275

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124987 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44656 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35393 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131836 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76954 "Built successfully") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45348 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53222 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95126 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7c95fb0b-31c6-457a-81a5-c0c5d45a544e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127941 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36190 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111776 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75673 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e4eb2a18-d54e-4acf-a831-5ea24b7bc7d5) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35117 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75317 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105953 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30160 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134510 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51810 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39611 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103603 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52235 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107989 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103379 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48721 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27007 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48836 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19595 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51694 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57489 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51068 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54425 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52760 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Source/WebGPU/WebGPU/Queue.mm:
```diff
@@ -505,12 +505,17 @@ static void invalidateCommandBuffers(Vector<Ref<WebGPU::CommandBuffer>>&& comman
     simd::ushort32 maxUshort = dataUshort.front();
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpsabi"
```
I guess this only matters on Intel x86_64 architectures with AVX?
<https://clang.llvm.org/docs/DiagnosticsReference.html#wpsabi>

ddkilzer commented on Source/WebKit/Configurations/BaseExtension.xcconfig:
```diff
@@ -25,6 +25,7 @@
 
 CLANG_ENABLE_OBJC_ARC = YES;
 CLANG_ENABLE_MODULES = YES;
+CLANG_MODULE_CACHE_PATH = ;
```
I think this is fine to land, but this seems like a workaround for a larger/deeper issue.  Maybe worth filing a radar about?

rniwa replied:
I think it's caused by some kind of mismatch between `xcodebuild` and `clang`. I don't think there is a real bug in Xcode / clang.

ddkilzer reviewed (approved):
r=me

rniwa committed:
WebKit LLVM branch cannot build WebKit for iOS simulator
https://bugs.webkit.org/show_bug.cgi?id=300293

Reviewed by David Kilzer and Mike Wyrzykowski.

Make WebKit compatible with webkit branch of LLVM.

* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::maxIndexValue): Ignore "-Wpsabi" warning.
* Source/WebKit/Configurations/BaseExtension.xcconfig:
Set CLANG_MODULE_CACHE_PATH to an empty string to avoid hitting
an error in pre-compiling WebKit2Prefix.h.

Canonical link: https://commits.webkit.org/301156@main
